### PR TITLE
feat: icp coin modularization

### PR DESCRIPTION
**Pull request by Canestin**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

ICP Coin Modularization

### ❓ Context

- **JIRA**: https://ledgerhq.atlassian.net/browse/LIVE-13680

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/ledger-live-common/src/families/internet_computer/setup.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { TransactionStatus, Transaction } from "@ledgerhq/coin-internet_computer/types/index";
+import { ICPGetAddrResponse, ICPSignature, ICPSigner } from "./types";
+import { getPath, isError } from "./utils";
+
+const createSigner: CreateSigner<ICPSigner> = (transport: Transport) => {
+  const icp = new ICP(transport);
+  return {
+    showAddressAndPubKey: async (path: string): Promise<ICPGetAddrResponse> => {
+      const r = await icp.showAddressAndPubKey(getPath(path));
+      isError(r);
```
Comment (qperrot) [inline libs/ledger-live-common/src/families/internet_computer/setup.ts:21]: `isError` is used only here, move the fonction here like in `filecoin` create a `common.ts`

File: libs/coin-modules/coin-internet_computer/src/test/cli.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import type { Account, AccountLike, AccountLikeArray } from "@ledgerhq/types-live";
+import invariant from "invariant";
+import flatMap from "lodash/flatMap";
+import type { Transaction } from "../types";
+
+function inferAccounts(account: Account): AccountLikeArray {
+  invariant(account.currency.family === "hedera", "hedera family");
```
[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/src/test/cli.ts]: [`cec1eed` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/cec1eed22c32c683480bab5577a59cf338f3044f#diff-29fe3a540f945ba9e3fda98ff0a462c8499b396f36d3ac567e697dc4d9ac1918R6-R7)

File: libs/coin-modules/coin-internet_computer/src/common-logic/index.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export {
+  isNoErrorReturnCode,
+  getPath,
+  isValidHex,
+  isValidBase64,
+  isError,
```
[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/src/common-logic/index.ts]: [`cec1eed` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/cec1eed22c32c683480bab5577a59cf338f3044f#diff-29fe3a540f945ba9e3fda98ff0a462c8499b396f36d3ac567e697dc4d9ac1918R6-R7)

[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/src/common-logic/index.ts]: [`cec1eed` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/cec1eed22c32c683480bab5577a59cf338f3044f#diff-f26ac1555cffea4a651d9f9233a7f48e96447e9756d692e3bc7be946a787f318L2-L6)

File: libs/coin-modules/coin-internet_computer/package.json [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      "require": "./lib/api/index.js",
+      "default": "./lib-es/api/index.js"
+    },
+    "./deviceTransactionConfig": {
+      "require": "./lib/bridge/deviceTransactionConfig.js",
+      "default": "./lib-es/bridge/deviceTransactionConfig.js"
+    },
+    "./erc20": {
+      "require": "./lib/erc20/tokenAccounts.js",
+      "default": "./lib-es/erc20/tokenAccounts.js"
```
[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/package.json]: [`414ed5f` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/414ed5f4f8fcfea3d789ff825a55c85045f2317b#diff-ccf2d5bf5f99467da28891a6bfe3990f18a48dedc7270201792bf41101fbe51aL54-L115)

File: libs/coin-modules/coin-internet_computer/package.json [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    },
+    "./logic": {
+      "require": "./lib/common-logic/index.js",
+      "default": "./lib-es/common-logic/index.js"
+    },
+    "./network": {
+      "require": "./lib/network/index.js",
+      "default": "./lib-es/network/index.js"
```
[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/package.json]: [`414ed5f` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/414ed5f4f8fcfea3d789ff825a55c85045f2317b#diff-ccf2d5bf5f99467da28891a6bfe3990f18a48dedc7270201792bf41101fbe51aL54-L115)

File: libs/coin-modules/coin-internet_computer/package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/package.json]: [`414ed5f` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/414ed5f4f8fcfea3d789ff825a55c85045f2317b#diff-ccf2d5bf5f99467da28891a6bfe3990f18a48dedc7270201792bf41101fbe51aL55-L58)

File: libs/coin-modules/coin-internet_computer/src/bridge/bridgeHelpers/icpRosetta/index.ts [hunk 6]
```diff
@@ -86,13 +88,17 @@ export const signICPTransaction = async ({
...
   });
 
-  const signedTxnRes = await icp.sign(path, Buffer.from(txnBlobToSign), 0);
-  isError(signedTxnRes);
+  const { r } = await signerContext(deviceId, async signer => {
+    const r = await signer.sign(path, Buffer.from(txnBlobToSign), ICP_SEND_TXN_TYPE);
+    return { r };
+  });
+
+  isError(r);
```
[the remark this answers is not in the document]
Comment (Canestin) [reply libs/coin-modules/coin-internet_computer/src/bridge/bridgeHelpers/icpRosetta/index.ts]: [`414ed5f` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/414ed5f4f8fcfea3d789ff825a55c85045f2317b#diff-4ac9a709b604dddf67b9ea73936eaeb728d5f84ba5e09bd35a5de140d5cf5498L96-L97)

File: libs/ledger-live-common/src/families/internet_computer/setup.ts (hunk 1 above)
Comment (Canestin) [reply libs/ledger-live-common/src/families/internet_computer/setup.ts:21]: [`414ed5f` (#8560)](https://github.com/LedgerHQ/ledger-live/pull/8560/commits/414ed5f4f8fcfea3d789ff825a55c85045f2317b#diff-3eb81eecd857d372d1d69641bee437b2aa79a282bca5a5178c16c4accc91d3ae)

File: libs/coin-modules/coin-internet_computer/src/bridge/signOperation.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import { SignerContext } from "@ledgerhq/coin-framework/signer";
+import { ICPSigner } from "../types";
+import { getPath } from "../common-logic";
+
+export const buildSignOperation =
+  (signerContext: SignerContext<ICPSigner>): AccountBridge<Transaction, Account>["signOperation"] =>
+  ({ account, transaction, deviceId }) =>
+    new Observable(o => {
+      async function main() {
+        // log("debug", "[signOperation] start fn");
```
Comment (qperrot) [inline libs/coin-modules/coin-internet_computer/src/bridge/signOperation.ts:21]: can be removed ?

File: libs/coin-modules/coin-internet_computer/src/common-logic/utils.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import { BigNumber } from "bignumber.js";
+import { ICP_SEND_TXN_TYPE, MAX_MEMO_VALUE } from "../consts";
+
+const validHexRegExp = new RegExp(/[0-9A-Fa-f]{6}/g);
+const validBase64RegExp = new RegExp(
+  /^(?:[A-Za-z\d+/]{4})*(?:[A-Za-z\d+/]{3}=|[A-Za-z\d+/]{2}==)?$/,
+);
+
+// export const isNoErrorReturnCode = (code: number): boolean => code === 0x9000;
```
Comment (qperrot) [inline libs/coin-modules/coin-internet_computer/src/common-logic/utils.ts:9]: Can be removed

**Review by qperrot — COMMENTED** [review]
Need to remove commented functions and add changeset

**Review by qperrot — APPROVED** [review]
LGM !

**Pull request merged**